server/src/main/java/io/seata/server/coordinator/DefaultCore.java

```diff
@@ -135,19 +131,24 @@ public GlobalStatus commit(String xid) throws TransactionException {
         if (globalSession == null) {
             return GlobalStatus.Finished;
         }
-        GlobalStatus status = globalSession.getStatus();
-
-        globalSession.closeAndClean(); // Highlight: Firstly, close the session, then no more branch can be registered.
-
-        if (status == GlobalStatus.Begin) {
-            globalSession.changeStatus(GlobalStatus.Committing);
-            if (globalSession.canBeCommittedAsync()) {
-                asyncCommit(globalSession);
-                return GlobalStatus.Committed;
-            } else {
-                doGlobalCommit(globalSession, false);
+        // just lock changeStatus
+        boolean shouldCommit = globalSession.lockAndExcute(() -> {
+            GlobalStatus status = globalSession.getStatus();
+            globalSession.closeAndClean(); // Highlight: Firstly, close the session, then no more branch can be registered.
+            if (status == GlobalStatus.Begin) {
```

Comment:
remove GlobalStatus status = globalSession.getStatus(); use globalSession.getStatus() to equal